secure-staff-services.md

```diff
@@ -0,0 +1,72 @@
+
+### Web Application Firewall (WAF)
+
+To provide a standard way to recognise and prevent common attacks than can come via web requests we will use AWS WAF [Web Access Control Lists (ACLs)](https://docs.aws.amazon.com/waf/latest/developerguide/web-acl.html). These can be integrated with various AWS solutions including AWS ALBs. See the following [Architecture Decision Record](https://github.com/guardian/waf/blob/main/adr/waf-management.md) for further context.
+
+This solution allows centrally managing Web ACL rules so that they are easily understood, contain an audit trail via source control and can easily be reviewed by InfoSec. In addition this solution can provide a [stream of request events](https://github.com/guardian/waf/blob/bd6fa3a45e9b1892b6207cdd6e8fff27930ca40a/lib/primary-waf.ts#L62) that can be centrally audited.
+
+**Your application should make use of a WAF with centrally managed Web ACL rules provisioned from the https://github.com/guardian/waf repository after review from a member of InfoSec.**
+
+### Authentication & Authorisation at the Application
+
+**Your application, either developed or off the shelf has authentication (verifies the identity of a user or service) and authorization (determines their access rights). The application must carry out these and not rely solely on authentication at the ALB.**
+
+You should use OpenID Connect (OIDC) with Google as an IdP to achieve this. The following mechanisms are acceptable:
+
+- [Verify any access token](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html) and any claims they contain passed from the ALB during the authentication step.
+- Authenticate and authorise the user in your application using an existing auth library such as:
+  - Scala: [guardian/play-googleauth](https://github.com/guardian/play-googleauth)
+  - Node: [guardian/pan-domain-authentication](https://github.com/guardian/pan-domain-authentication/#to-verify-login-in-nodejs). 
+
+### Authorisation / Access Management
+
+**You should follow the [Principle of Least Privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege), when deciding who has access to your application** and consider the sensitivity of any data or operations your service makes available. For example a service which provides read-only data on what's for lunch in the canteen may be accessed by all Guardian staff, but a service that publishes content to our site must be tightly controlled to a known set of users.
+
+Even if we establish the identity of a user using Google as our IdP, we must further establish that they are a Guardian staff member and in many cases as a member of a specific group of Guardian users.
+
+We can currently enforce as part of configuring an OAuth client in a Google Cloud project that any user authenticated with that client are an internal user, but cannot further restrict access to specific users or Google groups.
```

Comment:
I am not sure this is true - not every user with guardian.co.uk email address is internal user. We have number of vendor accounts.

Reply:
@stiliyana-simeonova Is there any way to differentiate those vendor accounts? Also do you know the use-case for creating vendor accounts?

Reply:
@stiliyana-simeonova i've updated language to reflect this. @markjamesbutler going to resolve this thread, suggest the question is answered out of band.